### PR TITLE
TAE:: support  for appending and committing a block whose data is already on  S3/FS.

### PR DESCRIPTION
**Pull request by triump2020**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #4693

## What this PR does / why we need it:
      1.  Support  for appending and committing a block whose data is already on  S3/FS.
       2.  When DN receives  precommit-write ,  it start to load Primary Keys(PK) of block asynchronously.
      3.   When txn at DN receives  prepare or commit , it wait PK  loading finished, and do deduplication
            through these keys  against  its snapshot , workspace, logtail.
TODO::
- [ ] need more UT for it.

### CODE REVIEW

File: pkg/catalog/types.go [hunk 1]
```diff
@@ -345,6 +360,13 @@ var (
 		types.New(types.T_TS, 0, 0, 0),                        // committs
 		types.New(types.T_uint64, 0, 0, 0),                    // segment_id
 	}
+
+	//for loading block on S3/FS
+	MoBlockMetaOnFSTypes = []types.Type{
```
Comment (nnsgmsone) [inline pkg/catalog/types.go:365]: What exactly does this schema represent?

**Review by ouyuanning — APPROVED** [review]

**Review by XuPeng-SH — CHANGES_REQUESTED** [review]
The design needs to be modified and then implemented after the design is completed. We'll discuss it offline

**Review by fengttt — COMMENTED** [review]
where is the conflict detection on primary keys for those records in the s3 block?

I assume this change handles transactions that write both s3 and records, plus delete, correctly

finally how do you test?

**Comment by triump2020** [comment]
> 

  The design needs to modify, close this PR now,  would  recreate a new PR !

**Pull request closed without merging**